dependencies.yaml

```diff
@@ -76,6 +75,33 @@ dependencies:
         packages:
           - wheel
           - setuptools>=61.0.0
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              arch: x86_64
+              cuda: "11.8"
+            packages:
+              - nvcc_linux-64=11.8
+          - matrix:
+              arch: aarch64
+              cuda: "11.8"
+            packages:
+              - nvcc_linux-aarch64=11.8
+          - matrix:
+              cuda: "12.0"
+            packages:
+              - cuda-version=12.0
+              - cuda-nvcc
+      - output_types: [conda, requirements, pyproject]
+        matrices:
+          - matrix:
+              cuda: "12.0"
+            packages:
+              - &cuda_python12 cuda-python>=12.0,<13.0
+          - matrix: # All CUDA 11 versions
```

Comment:
Not a blocking comment, but something we should address: this fallback will give incorrect results if someone tries CUDA>=12.1, which may start happening fairly soon (or perhaps in DLFW/some other container where people are using dfg, e.g. I know Paul is doing that for devcontainers in some cases).

Reply:
I can specify each CUDA 11 version here, to enforce a failure for CUDA 12.1 instead of a silent match for the wrong package.

Reply:
Turns out this is not possible because we need rapids-dependency-file-generator to support pyproject.toml with matrix keys. We have to have a "no matrix" fallback that is used for pyproject.toml generation.

Reply:
https://github.com/rapidsai/dependency-file-generator/issues/46

Reply:
Right, let's come back to this after we resolve the dfg work.